client/src/app/panel/Panel.js

```diff
@@ -144,40 +144,42 @@ function Tab(props) {
 
   const { addTab, removeTab } = useContext(TabContext);
 
-  const Link = <>
-    <span className="panel__link-label">
-      { label }
-    </span>
-    {
-      isDefined(number)
-        ? <span className="panel__link-number">{ number }</span>
-        : null
-    }
-  </>;
-
-  const Actions = actions.map(action => {
-    const Icon = action.icon;
-
-    return <button key={ action.title } className="panel__action" title={ action.title } onClick={ action.onClick }>
-      <Icon />
-    </button>;
-  });
-
-  const TabContent = {
-    id: id || label,
-    priority,
-    link: Link,
-    actions: Actions,
-    body: children
-  };
+  const TabContent = useMemo(() => {
```

Comment:
What happens here, and why?

Reply:
We recreate the TabContent every time a Prop changes (instead of every rerender). This allows us to react to changed content [here](https://github.com/camunda/camunda-modeler/blob/73623ed3c9d9a16b628af366890ed836b94ed330/client/src/app/panel/Panel.js#L182) and update the tab in the `TabContext`

Reply:
Using `useMemo` here prevents unnecessary updates of the `TabContext`

Reply:
Makes sense now. So it broke turning it into a functional component.